internal/services/instances.go

```diff
@@ -133,3 +128,81 @@ func (s *ServiceClient) DeleteAll(ctx context.Context, targetNamespace string) e
 
 	return errors.Wrap(err, "error deleting helm charts")
 }
+
+func (s *ServiceClient) List(ctx context.Context, namespace string) ([]*models.Service, error) {
+	serviceList := []*models.Service{}
+
+	listOpts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf(
+			"%s,%s,%s=%s",
+			ServiceNameLabelKey,
+			CatalogServiceLabelKey,
+			TargetNamespaceLabelKey, namespace,
+		),
+	}
+
+	unstructuredServiceList, err := s.helmChartsKubeClient.Namespace(helmchart.Namespace()).List(ctx, listOpts)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return serviceList, nil
+		}
+		return nil, errors.Wrap(err, "fetching the service instance")
+	}
+
+	helmChartList, err := convertUnstructuredListIntoHelmCharts(unstructuredServiceList)
+	if err != nil {
+		return nil, errors.Wrap(err, "error converting unstructured list to helm charts")
+	}
+
+	for _, srv := range helmChartList {
```

Comment:
This loop is making 2N+1 queries to the kube API (1 to list all helm charts and then 1 queries per helmchart to find the catalog service and the status). I'm pretty sure this will perform [worse than the namespace listing](https://github.com/epinio/epinio/issues/1353). 

On small improvement would be to fetch all catalog services once, create a lookup map and check against that for existence of the catalog service. This will still be N+2 queries though. If we skip the status of the services in the list view we go down to 2 queries (one for service + one for catalog services).

It would be nice to see the status of the helm release in the listing page but not against performance. I think it should be the helm-controller's job to update the helmchart status to match the one of the helm release (maybe it does?).

Reply:
I've pushed the small improvement that will fetch all the services and then will look for the `Missing` services with a lookup map. I've left the release status, I'm not sure we are going to have many services running in the same workspace, we can drop the status or see if it's going to be a problem in the future.

Reply:
Let's keep it as it is for now